mono/metadata/domain-internals.h

```diff
@@ -454,6 +455,12 @@ struct _MonoDomain {
 	gboolean throw_unobserved_task_exceptions;
 
 	guint32 execution_context_field_offset;
+
+#ifdef ENABLE_NETCORE
+	GSList *alcs;
```

Comment:
Why is this stored in MonoDomain, when there is only one domain in netcore ?

Reply:
I could make it global, if you prefer.

I was hoping to also modularize non-netcore Mono's loader using ALCs in some follow-up work.  (They wouldn't have managed counterparts, just internal to the runtime).  Would be less of a hack for `LoadFile`, `LoadFrom` etc.

Reply:
Oh, one other thing: it makes it easier to catch lifecycle issues.

The default ALC gets created when the root domain is created, if you don't have a domain yet, you can't get the default ALC either.

And if you're using the embedding API and try to get the default ALC from a thread that's not attached, you can't do it.